infoschema/tables.go

```diff
@@ -867,24 +867,14 @@ func (c *statsCache) get(ctx sessionctx.Context) (map[int64]uint64, map[tableHis
 }
 
 func getAutoIncrementID(ctx sessionctx.Context, schema *model.DBInfo, tblInfo *model.TableInfo) (int64, error) {
-	hasAutoIncID := false
-	for _, col := range tblInfo.Cols() {
-		if mysql.HasAutoIncrementFlag(col.Flag) {
-			hasAutoIncID = true
-			break
-		}
+	is := ctx.GetSessionVars().TxnCtx.InfoSchema.(InfoSchema)
+	tbl, err := is.TableByName(schema.Name, tblInfo.Name)
+	if err != nil {
+		return 0, err
 	}
-	autoIncID := tblInfo.AutoIncID
-	if hasAutoIncID {
-		is := ctx.GetSessionVars().TxnCtx.InfoSchema.(InfoSchema)
-		tbl, err := is.TableByName(schema.Name, tblInfo.Name)
-		if err != nil {
-			return 0, err
-		}
-		autoIncID, err = tbl.Allocator(ctx).NextGlobalAutoID(tblInfo.ID)
-		if err != nil {
-			return 0, err
-		}
+	autoIncID, err := tbl.Allocator(ctx).NextGlobalAutoID(tblInfo.ID)
```

Comment:
Could we use the nextAutoID in this allocator of this table, in the memory

Reply:
But ` select table_name, auto_increment from NFORMATION_SCHEMA.TABLES` from different-tidb-server will got different value.

Reply:
done.